src/contexts/CommentContext.js

```diff
@@ -0,0 +1,57 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { usePosts } from "./PostContext";
+
+
+const CommentContext = createContext()
+
+export const useComments = () => useContext(CommentContext)
+
+const CommentProvider = ({children}) => {
+
+    const [comments,setComments] = useState([])
+
+
+    const fetchAndSetComments =  async (postId) => {
+        const response = await fetch(`https://jsonplaceholder.typicode.com/posts/${postId}/comments`)
+        const data = await response.json()
+        localStorage.setItem('comments',JSON.stringify(data))
+        setComments(data)
+    }
+
+    useEffect(()=>{
+        localStorage.setItem('comments',comments)
+    },[comments])
+
+    useEffect(()=>{
+        let json = localStorage.getItem('comments')
+        if(json){
+            setComments(JSON.parse(json))
+        }
+    },[])
+
+    const addComment = (comment) => {        
```

Comment:
remove brackets around comment as there is no need of that if there is only one prop